core/src/main/java/io/substrait/relation/RelProtoConverter.java

```diff
@@ -228,6 +237,33 @@ public Rel visit(ExtensionTable extensionTable) throws RuntimeException {
     return Rel.newBuilder().setRead(builder).build();
   }
 
+  @Override
+  public Rel visit(HashJoin hashJoin) throws RuntimeException {
+    var builder =
+        HashJoinRel.newBuilder()
+            .setCommon(common(hashJoin))
+            .setLeft(toProto(hashJoin.getLeft()))
+            .setRight(toProto(hashJoin.getRight()))
+            .setType(hashJoin.getJoinType().toProto());
+
+    List<FieldReference> leftKeys = hashJoin.getLeftKeys();
+    List<FieldReference> rightKeys = hashJoin.getRightKeys();
+
+    if (leftKeys.size() != rightKeys.size()) {
+      throw new RuntimeException("Number of left and right keys must be equal.");
+    }
+
+    for (int idx = 0; idx < hashJoin.getLeftKeys().size(); idx++) {
+      builder.setLeftKeys(idx, toProto(leftKeys.get(idx)));
+      builder.setRightKeys(idx, toProto(rightKeys.get(idx)));
+    }
```

Comment:
@vbarua can I do better here? 
Also validation of left and right key sizes is a bit doubtful.

Reply:
The validation seems reasonable to me. The only thing I would suggest to improve it would be to provide additional context. Something like:
```
The number of left and right keys must be equal for HashJoin. Received left %s and right %s keys.
```

For setting the keys themselves, I think
```java
    builder.addAllLeftKeys(leftKeys.stream().map(this::toProto).collect(Collectors.toList()));
    builder.addAllRightKeys(rightKeys.stream().map(this::toProto).collect(Collectors.toList()));
```
is a bit cleaner.

Reply:
This is perfect. I will update.

Reply:
Updated.